crates/ruff/src/settings/types.rs

```diff
@@ -60,6 +83,21 @@ impl PythonVersion {
             Self::Py311 => (3, 11),
         }
     }
+
+    pub fn get_minimum_supported_version(requires_version: &RequiresVersion) -> Option<Self> {
+        let mut minimum_version = None;
+        for python_version in PythonVersion::iter() {
+            if requires_version
+                .0
+                .iter()
+                .all(|specifier| specifier.contains(&python_version.into()))
```

Comment:
Is this going to do the right thing if `pyproject.toml` specifies a patch release like `requires-python = ">= 3.8.16"`?

Reply:
No, it will return `py39`.

Reply:
This is fixed, I hard coded the patch number to 100 to avoid special casing.

Reply:
i tried to come up with something better but i agree a really high patch number is the really the best we can do here